### PR TITLE
Add notebook that explains neural-lam pytorch tensors graph disk format

### PR DESCRIPTION
**Pull request by leifdenby**
## Describe your changes

Add notebook that describes the current `neural-lam` pickled pytorch tensors graph disk format. I feel like this documentation is currently missing, and since we write to this format in `weather-model-graphs` I think this would be a place where we could document it. This documentation could maybe also exist within `neural-lam` itself. But I feel like having this notebook could be a starting point at least.

**NOTE**: The notebook also shows that there is currently a bug in the index ordering in adjacency lists stored into pickle pytorch Tensors. 

No change of dependencies needed.

## Issue Link

n/a

## Type of change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📖 Documentation (Addition or improvements to documentation)

## Checklist before requesting a review

- [ ] My branch is up-to-date with the target branch - if not update your fork with the changes from the target branch (use `pull` with `--rebase` option if possible).
- [ ] I have performed a self-review of my code
- [ ] For any new/modified functions/classes I have added docstrings that clearly describe its purpose, expected inputs and returned values
- [ ] I have placed in-line comments to clarify the intent of any hard-to-understand passages of my code
- [ ] I have updated the documentation to cover introduced code changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have given the PR a name that clearly describes the change, written in imperative form ([context](https://www.gitkraken.com/learn/git/best-practices/git-commit-message#using-imperative-verb-form)).
- [ ] I have requested a reviewer and an assignee (assignee is responsible for merging)

## Checklist for reviewers

Each PR comes with its own improvements and flaws. The reviewer should check the following:
- [ ] the code is readable
- [ ] the code is well tested
- [ ] the code is documented (including return types and parameters)
- [ ] the code is easy to maintain

## Author checklist after completed review

- [ ] I have added a line to the CHANGELOG describing this change, in a section
  reflecting type of change (add section where missing):
  - *added*: when you have added new functionality
  - *changed*: when default behaviour of the code has been changed
  - *fixes*: when your contribution fixes a bug

## Checklist for assignee

- [ ] PR is up to date with the base branch
- [ ] the tests pass
- [ ] author has added an entry to the changelog (and designated the change as *added*, *changed* or *fixed*)
- Once the PR is ready to be merged, squash commits and merge the PR.

### CODE REVIEW

**Comment by joeloskarsson** [comment]
This should be very useful indeed! Looking at the output there seems to a bug yes. It could very well be the label sorting that's missing, that you linked to as fixed in #34. Have you tried rebasing this on top of that PR and see if you have the same issue?

File: docs/saving_graphs_for_neural_lam.ipynb [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+    "import matplotlib.pyplot as plt\n",
+    "import torch\n",
+    "import networkx as nx"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will start by creating some a graph from some randomly generated latitudes and longitudes:"
```
Comment (joeloskarsson) [inline docs/saving_graphs_for_neural_lam.ipynb]: When we feed coordinates like this, without a specified CRS, these are not necessarily assumed to be lat-lons.
```suggestion
    "We will start by creating some a graph from some randomly generated coordinates:"
```

**Comment by leifdenby** [comment]
Since all subgraphs now use zero-offset indexing (i.e. both the source and destination sets of nodes have indexes running from `0` and to the number of nodes), and we now have a detailed spec included with neural-lam: https://github.com/mllam/neural-lam/blob/main/docs/graph_storage_spec.md I am going to close this PR. I don't think it is necessary anymore. People should look at that spec and the associated validator for details about the graph format on disk that `neural-lam` expects :)
